wpilibNewCommands/src/main/native/cpp/frc2/command/CommandScheduler.cpp

```diff
@@ -450,36 +450,40 @@ void CommandScheduler::RequireUngrouped(
   }
 }
 
-void CommandScheduler::InitSendable(nt::NTSendableBuilder& builder) {
+void CommandScheduler::InitSendable(wpi::SendableBuilder& builder) {
   builder.SetSmartDashboardType("Scheduler");
-  builder.SetUpdateTable(
-      [this,
-       namesPub = nt::StringArrayTopic{builder.GetTopic("Names")}.Publish(),
-       idsPub = nt::IntegerArrayTopic{builder.GetTopic("Ids")}.Publish(),
-       cancelEntry = nt::IntegerArrayTopic{builder.GetTopic("Cancel")}.GetEntry(
-           {})]() mutable {
-        auto toCancel = cancelEntry.Get();
-        if (!toCancel.empty()) {
-          for (auto cancel : cancelEntry.Get()) {
-            uintptr_t ptrTmp = static_cast<uintptr_t>(cancel);
-            Command* command = reinterpret_cast<Command*>(ptrTmp);
-            if (m_impl->scheduledCommands.find(command) !=
-                m_impl->scheduledCommands.end()) {
-              Cancel(command);
-            }
-          }
-          cancelEntry.Set({});
-        }
-
-        wpi::SmallVector<std::string, 8> names;
-        wpi::SmallVector<int64_t, 8> ids;
+  builder.AddStringArrayProperty(
+      "Names",
+      [this]() mutable {
+        std::vector<std::string> names;
         for (Command* command : m_impl->scheduledCommands) {
           names.emplace_back(command->GetName());
+        }
+        return names;
+      },
+      nullptr);
+  builder.AddIntegerArrayProperty(
+      "Ids",
+      [this]() mutable {
+        std::vector<int64_t> ids;
+        for (Command* command : m_impl->scheduledCommands) {
           uintptr_t ptrTmp = reinterpret_cast<uintptr_t>(command);
           ids.emplace_back(static_cast<int64_t>(ptrTmp));
         }
-        namesPub.Set(names);
-        idsPub.Set(ids);
+        return ids;
+      },
+      nullptr);
+  builder.AddIntegerArrayProperty(
+      "Cancel", []() { return std::vector<int64_t>{}; },
```

Comment:
I don't think this replicates the old behavior.  The old behavior looped over the array and then set the value to an empty array.  This continuously sets the value to an empty array.

Reply:
I added a test to ensure this still worked, and it passes in both `main` and this PR. My thought process for the implementation was that because setters are called first (which pulls in updates from something like a dashboard), the setter code gets run, cancelling any commands in the entry. That's not actually why this works. Since the subscriber excludes updates from the publisher, if a dashboard updates the `Cancel` entry to cancel a command, even if the getter is called first and replaces the entry with an empty array, the update will still go through, and the command gets canceled. The execution order might change depending on if getters or setters are called first, but it shouldn't make a difference.